src/main/java/com/google/devtools/build/lib/analysis/actions/StrippingPathMapper.java

```diff
@@ -0,0 +1,268 @@
+ * effects:
+ *
+ * <ol>
+ *   <li>"Qualifying" actions strip config paths from their command lines. An action qualifies if
+ *       its implementation logic uses {@link PathMappers#create(Action, OutputPathsMode)} as
+ *       described in its javadocs and has its mnemonic listed in {@link
+ *       PathMappers#SUPPORTED_MNEMONICS}. Such an action must pass the {@link PathMapper} to all
+ *       structured command line constructions. If any unstructured command line arguments refer to
+ *       artifact paths, custom handling needs to be added to {@code mapCustomStarlarkArgv} or
+ *       {@code getMapFn} below.
+ *   <li>A supporting executor strips paths from qualifying actions' inputs and outputs before
+ *       staging for execution by taking {@link Spawn#getPathMapper()} into account.
+ * </ol>
+ *
+ * <p>So an action is responsible for declaring that it strips paths and adjusting its command line
+ * accordingly. The executor is responsible for remapping action inputs and outputs to match.
+ */
+public final class StrippingPathMapper {
+  static final String GUID = "8eb2ad5a-85d4-435b-858f-5c192e91997d";
+
+  /**
+   * Creates a new {@link PathMapper} that strips config prefixes if the particular action instance
+   * supports it.
+   *
+   * @param action the action to potentially strip paths from
+   * @return a {@link StrippingPathMapper} if the action supports it, else {@link Optional#empty()}.
+   */
+  static Optional<PathMapper> tryCreate(Action action) {
+    // This is expected to always be "bazel-out", but we don't want to hardcode it here.
+    PathFragment outputRoot = action.getPrimaryOutput().getExecPath().subFragment(0, 1);
+    if (isPathStrippable(action.getInputs(), outputRoot)) {
+      return Optional.of(
+          create(action.getMnemonic(), action instanceof StarlarkAction, outputRoot));
+    }
+    return Optional.empty();
+  }
+
+  private static PathMapper create(String mnemonic, boolean isStarlarkAction,
+      PathFragment outputRoot) {
+    final StringStripper argStripper = new StringStripper(outputRoot.getPathString());
+    final ExceptionlessMapFn<Object> structuredArgStripper =
+        (object, args) -> {
+          if (object instanceof String) {
+            args.accept(argStripper.strip((String) object));
+          } else {
+            args.accept(CommandLineItem.expandToCommandLine(object));
+          }
+        };
+    // This kind of special handling should not be extended. It is a hack that works around a
+    // limitation of the native implementation of location expansion: The output is just a list of
+    // strings, not a structured command line that would allow transparent path mapping.
+    // Instead, reimplement location expansion in Starlark and have it return an Args object.
+    final boolean isJavaAction =
+        mnemonic.equals("Javac") || mnemonic.equals("JavacTurbine") || mnemonic.equals("Turbine");
```

Comment:
@fmeum I'm seeing `JavaResourceJar` mnemonic is not marked as Java action due to this check. The consequence of that is when the code path reaches `CustomCommandLine.evalSimpleVectorArg` and it's not an ActionInput, the mapFn is not return resulting in arguments being unmapped.
![Screenshot 2023-09-27 at 2 05 10 PM](https://github.com/bazelbuild/bazel/assets/91614797/c4d1ffa9-a0ee-43f8-b3ce-8c51dd587a1e)

so the fix works with internal test is adding `JavaResourceJar` mnemonic in the check here and make `PathMapper.getMapFn` to do the following
```
if (isJavaAction ){
          if ("--javacopts".equals(previousFlag) || "--resources".equals(previousFlag)) {
            return structuredArgStripper;
          }
        }
        return MapFn.DEFAULT;
```

In your comment in this file line 108, you mentioned this check shouldn't be extended so I want to get your thoughts on doing this.